.dotnet/src/Utility/AsyncServerSentEventEnumerable.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace OpenAI;
+
+/// <summary>
+/// Represents a collection of SSE events that can be enumerated as a C# async stream.
+/// </summary>
+internal class AsyncServerSentEventEnumerable : IAsyncEnumerable<ServerSentEvent>
```

Comment:
These are all pulled over from https://github.com/Azure/azure-sdk-for-net/pull/43840